aggregate-counter-sink/src/test/java/org/springframework/cloud/stream/module/AggregateCounterTests.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.Collections;
+
+import org.joda.time.DateTime;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.module.metrics.AggregateCounter;
+import org.springframework.cloud.stream.module.metrics.AggregateCounterRepository;
+import org.springframework.cloud.stream.module.metrics.AggregateCounterResolution;
+import org.springframework.cloud.stream.module.metrics.AggregateCounterSink;
+import org.springframework.cloud.stream.module.metrics.AggregateCounterSinkApplication;
+import org.springframework.cloud.stream.test.junit.redis.RedisTestSupport;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+
+/**
+ * Tests the aggregate-counter module.
+ *
+ * @author Eric Bottard
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = AggregateCounterSinkApplication.class)
+@DirtiesContext
```

Comment:
Does it, really? Given that we reset the repository, seems like we can re-user the AppContext